### PR TITLE
Ready-to-use sidecar configurations

### PR DESCRIPTION
**Pull request by AntonEbel**
## Description

- add graylog_host variable and use it in templates
- add default configurations for windows and linux with `default` tag
- add Auditbeat collector for Linux

part of #15570

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java [hunk 1]
```diff
@@ -108,107 +135,139 @@ public void upgrade() {
...
+                                tags:
+                                 - windows
+                                winlogbeat:
+                                  event_logs:
+                                   - name: Application
+                                   - name: System
+                                   - name: Security""",
+                        beatsPreambel
+                )
+        ).ifPresent(collector -> ensureDefaultConfiguration("default-windows", collector));
```
Comment (mpfz0r) [inline graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java:200]:
```suggestion
        ).ifPresent(collector -> ensureDefaultConfiguration("winlogbeat-default", collector));
```
I'd rather name that after the collector. the OS is implicit. actually the collector is too, but I think it looks better this way

File: graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java [hunk 2]
```diff
@@ -108,107 +135,139 @@ public void upgrade() {
...
+                                  - /sbin
+                                  - /usr/sbin
+                                  - /etc
+                                output.logstash:
+                                   hosts: ["${user.graylog_host}:5044"]
+                                path:
+                                  data: /var/lib/graylog-sidecar/collectors/auditbeat/data
+                                  logs: /var/lib/graylog-sidecar/collectors/auditbeat/log""",
+                        beatsPreambel)
+        ).ifPresent(collector -> ensureDefaultConfiguration("default-linux", collector));
```
Comment (mpfz0r) [inline graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java:176]:
```suggestion
        ).ifPresent(collector -> ensureDefaultConfiguration("auditlogbeat-default", collector));
```
I'd rather name that after the collector. the OS is implicit. actually the collector is too, but I think it looks better this way

File: graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java [hunk 3]
```diff
@@ -89,17 +113,20 @@ public void upgrade() {
...
+                                  paths:
+                                    - /var/log/*.log
+                                  type: log
+                                output.logstash:
+                                   hosts: ["${user.graylog_host}:5044"]
+                                path:
+                                  data: ${sidecar.spoolDir!"/var/lib/graylog-sidecar/collectors/filebeat"}/data
+                                  logs: ${sidecar.spoolDir!"/var/lib/graylog-sidecar/collectors/filebeat"}/log""",
+                        beatsPreambel
+                )
```
Comment (mpfz0r) [inline graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java:129]: no default config for linux filebeat?

**Comment by mpfz0r** [comment]
apart from my comments this LGTM. I guess you wanted to extend the configs with the feedback from sec-dev?

File: graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java [hunk 4]
```diff
@@ -328,48 +393,93 @@ private void removeConfigPath() {
...
+            LOG.info("'{}' sidecar default configuration is missing, adding it.", name);
+            try {
+                config = configurationService.save(Configuration.createWithoutId(
+                        collector.id(),
+                        name,
+                        "#ffffff",
+                        collector.defaultTemplate(),
+                        Set.of("default")));
+            } catch (Exception e) {
+                LOG.error("Can't save sidecar default configuration '{}', please restart Graylog to fix this.", name, e);
```
[the remark this answers is not in the document]
Comment (AntonEbel) [reply graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java]: removed the "please restart Graylog to fix this." part.

**Comment by AntonEbel** [comment]
We have discussed with @kroepke  that we will continue without a new Winlogbeat configuration.

**Review by mpfz0r — APPROVED** [review]

File: graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java [hunk 5]
```diff
@@ -328,48 +470,93 @@ private void removeConfigPath() {
...
-        return collector.id();
+        return Optional.of(collector);
+    }
+
+    private void ensureConfigurationVariable(String name, String description, String content) {
+        ConfigurationVariable variable = null;
+        try {
+            variable = configurationVariableService.findByName(name);
+            if (variable == null) {
+                LOG.error("Couldn't find sidecar configuration variable '{}' fixing it.", name);
```
Comment (mpfz0r) [inline graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java:518]:
```suggestion
                LOG.info("Couldn't find sidecar configuration variable '{}' fixing it.", name);
```
This is not an error

**Review by mpfz0r — CHANGES_REQUESTED** [review]
one nitpick rest looks good

**Review by mpfz0r — APPROVED** [review]

**Pull request merged**